0025-pre-shared-key-2/0025-pre-shared-key-2.md

```diff
@@ -186,58 +156,59 @@ The following is encrypted, as defined [above](#encryption), and set as the ILP
+#### Request
 
-#### Request Fields
+PSK Request packets should ONLY be sent in ILP Prepare packets.
 
 | Field | Type | Description |
 |---|---|---|
-| Type (T) | 1-Byte UInt (UInt8) | `0` to indicate a payment chunk, `1` to indicate it is the last chunk for the payment. Receivers SHOULD NOT fulfill additional chunks for the given payment after the last chunk has been received |
-| Payment ID | 16-Byte UInt | Unique ID of this payment |
-| Sequence | 4-Byte UInt (UInt32) | Zero-indexed chunk sequence number. Receivers SHOULD NOT fulfill two chunks with the same sequence for a single payment. Receivers SHOULD use the sequence to order chunks in case application data is split across multiple chunks, but chunks do not necessarily need to be delivered in order |
-| Payment Amount (Total)* | 8-Byte UInt (UInt64) | The total amount the payment is intended to deliver. A value of `0` in a request indicates that the sender will determine when to stop sending chunks |
-| Chunk Amount (Minimum) | 8-Byte UInt (UInt64) | Minimum amount the receiver should receive for this chunk in order to fulfill it. A value of `0` in a request indicates that any amount should be accepted |
-| Application Data | [OER Variable-Length Octet String](http://www.oss.com/asn1/resources/books-whitepapers-pubs/Overview%20of%20OER.pdf) | User data carried along with the payment |
-| Extensions (E) | 1-Byte UInt (UInt8) | Currently set to `0`. Reserved for future use to indicate the presence of extension fields. Parsers that do not understand the extensions MUST ignore them |
-| Junk Data | 0-? Bytes | _(Optional)_ Extra data included to obscure what the ILP payment is for. Receivers MUST ignore this data. Senders SHOULD include either the maximum or a random number of empty (0) bytes |
+| Type (T) | Always `4` (UInt8) | Indicates that this is a PSK2 request. Request packets SHOULD be rejected or ignored if they are not in an ILP Prepare packet. |
+| Request ID (ReqID) | UInt32 | ID used to correlate requests and responses. Senders CANNOT rely on receivers enforcing uniqueness, but they SHOULD ensure that responses carry the same Request ID as the outgoing request |
+| Request Amount | UInt64 | Minimum amount that should arrive in the ILP Prepare packet for the receiver to fulfill it. Receivers SHOULD NOT accept packets where the Request Amount is greater than the ILP Prepare amount |
+| Data | [OER Variable-Length Octet String](http://www.oss.com/asn1/resources/books-whitepapers-pubs/Overview%20of%20OER.pdf) | User data carried along with the payment. Note this data is encrypted and authenticated |
+| Extensions | N/A | Addtional fields may be added after the `Application Data` later. Implementations that do not support any extensions SHOULD ignore all additional bytes |
+| Junk Data | 0-? Bytes | _(Optional)_ Extra data included to obscure what the ILP payment is for. Receivers SHOULD ignore this data |
```

Comment:
Receivers should ignore Extensions/Junk, but connectors are supposed to relay them, right (referencing https://github.com/interledgerjs/ilp-connector/issues/405).

Reply:
> Receivers should ignore Extensions/Junk

Yes

> but connectors are supposed to relay them, right?

This is part of the encrypted PSK packet so connectors won't know whether or not it's there. This doesn't use the extensions mentioned in https://github.com/interledgerjs/ilp-connector/issues/405